server/datastore/mysql/hosts.go

```diff
@@ -848,6 +848,12 @@ func (ds *Datastore) ListHosts(ctx context.Context, filter fleet.TeamFilter, opt
 
 	sql += hostMDMSelect
 
+	if opt.ListOptions.OrderKey == "last_restarted" {
+		sql += `,
+		-FLOOR(UNIX_TIMESTAMP() * 1000 - (((UNIX_TIMESTAMP() - UNIX_TIMESTAMP(detail_updated_at)) * 1000) + (uptime / 1000000))) AS last_restarted
```

Comment:
How about using `DATE_ADD` (we use that in many queries), something like:
```
select DATE_SUB(detail_updated_at, INTERVAL uptime/1000 MICROSECOND) as last_restarted from hosts;
```
Such query matches the date the UI renders in the "Last restarted" column.

We need to account for hosts that have `detail_updated_at=NULL` and/or `uptime=0` (default values in case the host hasn't reported any data yet).

Reply:
I implemented this just now, but have a question: let's say Host A has last restarted of 4 days, and Host B has last restarted of 6 days. When we sort by `desc`, do we expect

1. Host B, Host A
or
2. Host A, Host B?

Asking because to me option 1 makes more sense, but with the implementation above we get option 2. Fwiw, the Last Fetched column does option 2, so could be that I'm just misunderstanding the intended functionality.

Reply:
I'll defer to @noahtalerman @rachaelshaw 

(We could also change the UI to send a different `order_direction` on the first click to sort?)

Reply:
Oh yeah, several fixes we could do.

Reply:
@jahzielv is the issue here that the sort is based on the timestamp but the UI displays it in timeago format? If so I think @lucasmrod's suggestion about handling the sort differently in the UI makes sense here (because in the API I would expect descending to list the more recent dates first, but in the UI I would expect descending to list the longest time ago first)

Reply:
@rachaelshaw yes, I believe that is the issue. 

> handling the sort differently in the UI

so should we do something like send `desc` when we mean `asc` and vice versa? Another thing we could do would be to just invert the list in the UI code when sorting on this column. Would it be better to simply have the API match the UI's expectation to avoid confusion?

Reply:
Assuming this is the stable API (not the API for contributors): the first one, we should have the UI reverse when it sends "desc" and "asc". (Otherwise, for pure API users, the sort wouldn't make as much sense.)

If it's the API for contributors, might be easier to just reverse it on the back-end and leave a comment explaining why.

Reply:
I believe that this is the stable API. I'll implement the first option.